PyMca5/tests/ut/test_q_tiled_catalog_selector_dialog.py

```diff
@@ -0,0 +1,87 @@
+import enable_pymca_import  # noqa: F401
+
+from unittest.mock import Mock, patch
+
+import pytest
+from pytestqt.qtbot import QtBot
+
+from PyQt5.QtCore import QEvent
+from PyQt5.QtWidgets import QApplication
+
+from PyMca5.PyMcaGui.io.TiledCatalogSelector import TiledCatalogSelector
+from PyMca5.PyMcaGui.io.QTiledCatalogSelectorDialog import QTiledCatalogSelectorDialog
+
+
+@pytest.fixture
+def dialog_model(qapp: QApplication):
+    model = TiledCatalogSelector(parent=qapp)
+    yield model
+
+
+def test_init(qtbot: QtBot, dialog_model: TiledCatalogSelector):
+    """Can create a QTiledCatalogSelectorDialog object."""
+    QTiledCatalogSelectorDialog(model=dialog_model)
+
+
+def test_render(qtbot: QtBot, dialog_model: TiledCatalogSelector):
+    """Can render a QTiledCatalogSelectorDialog window."""
+    dialog = QTiledCatalogSelectorDialog(model=dialog_model)
+    dialog.show()
+    qtbot.addWidget(dialog)
+
+# Functional tests...
+
+def test_connection(qtbot: QtBot, dialog_model: TiledCatalogSelector):
+    """Verify changes enacted by initiating a connection."""
+    model = dialog_model
+    expected_url = "New URL"
+
+    model.url = expected_url
+    dialog = QTiledCatalogSelectorDialog(model=model)
+    dialog.show()
+    qtbot.addWidget(dialog)
+
+    with patch.object(model, "client_from_url") as mock_client_constructor:
+        client = Mock()
```

Comment:
It looks like `Mock` is used for two types of objects: Qt Events and Tiled clients. I have read the advice, "Only mock your own objects," presumably because third-party object may evolve under you without your knowledge or control.

I think it's pretty safe to mock a Qt Event because it's extremely stable and simple. I would actually worry more about mocking the Tiled client. If you want a real Tiled client fixture, here's one:

```py
from tiled.catalog import in_memory
from tiled.server.app import build_app
from tiled.client import Context, from_context

@pytest.fixture
def client(tmpdir):
    catalog = in_memory(writable_storage=tmpdir)
    app = build_app(catalog)
    with Context.from_app(app) as context:
        client = from_context(context)
        yield client
````

Reply:
I'll have to think about this one. The main point of _this mock_ is to have an object with the members `uri` and `api_uri` defined.  If the idea is not to rely on that interface, then I would probably be better off encapsulating it...and them mocking the encapsulation.

For any of the yet-to-be-written functional tests that depend on tiled client functionality, I will definitely want to use the in_memory client.  I might already want to do that for testing connection errors.

Reply:
Going to leave this as is for now.  Happy to revisit later as tiled client usage is expanded in this model class.